v2/pkg/buildassets/buildassets.go

```diff
@@ -124,15 +122,16 @@ func resolveProjectData(content []byte, projectData *project.Project) ([]byte, e
 }
 
 func writeFileSystemFile(projectData *project.Project, file string, content []byte) error {
-	path := filepath.Clean(filepath.Join(projectData.Path, filepath.FromSlash(file)))
+	baseDir := filepath.Dir(projectData.GetBuildDir())
+	targetPath := filepath.Clean(filepath.Join(baseDir, filepath.FromSlash(file)))
```

Comment:
With the above changed code, file should always be a slash path relative to BuildDir, in which case the targetPath could just be: `targetPath := filepath.Join(projectData.GetBuildDir(), filepath.FromSlash(file))`